framework/src/outputs/Checkpoint.C

```diff
@@ -75,60 +77,74 @@ Checkpoint::directory() const
 void
 Checkpoint::output(const ExecFlagType & /*type*/)
 {
-  // Create the output directory
-  std::string cp_dir = directory();
-  Utility::mkdir(cp_dir.c_str());
+  //check periodically if we should be outputting or not
+  if (!_should_output){
+    _should_output = Moose::autosave_flag;
```

Comment:
This would be the right place to output to the `_console`. Let's say we caught a signal and are addressing it here. From that standpoint, it might be handy to make your global variable an enum or bitmask instead of a Boolean. Perhaps we want to make it possible to capture different signals in the future and they could just be stored in a single flag "register".